### PR TITLE
fix(cli): changed the CLI template options and added support for federation

### PR DESCRIPTION
**Pull request by riginoommen**
# Resolves ONEPLAT-189
## Explain the feature/fix:

- Included `cookie parser` with graphql template

- Fixed the indendation of code in mongo db support

- Binded the native route as the first route for the rest based service

- Included the apollo federation support in the CLI

## Does this PR introduce a breaking change?

> No

## Screenshots:

<details>
<summary>View Screenshots</summary>

</details>

### Ready-for-merge Checklist

- [x] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)?
- [x] Does the change have appropriate unit tests?
- [x] Did tests pass?
- [x] Did you update or add any necessary documentation (README.md, WHY.md, etc.)?

### CODE REVIEW

**Comment by sayak-sarkar** [comment]
@riginoommen Can you please update the PR description to mention all the issues that this resolves?

File: cli-init.sh [hunk 1]
```diff
@@ -2,4 +2,4 @@
 cd packages
```
Comment (deshmukhmayur) [inline cli-init.sh:2]: Do you need to change directory inside the script?
Also, if you _have_ to change the directory, make sure you change it back to the present working directory after the script is executed.

File: cli-init.sh [hunk 2]
```diff
@@ -2,4 +2,4 @@
...
 
 # Initate template QA
-yo --no-insight --no-update-notifier ../cli/generator-op-microservice/generators/microservice
+../cli/generator-op-microservice/node_modules/yo/lib/cli.js --no-insight --no-update-notifier ../cli/generator-op-microservice/generators/microservice
```
Comment (deshmukhmayur) [inline cli-init.sh:5]: npm automatically creates a symlink in `node_modules/.bin` directory for all the executable npm commands.
```suggestion
../cli/generator-op-microservice/node_modules/.bin/yo --no-insight --no-update-notifier ../cli/generator-op-microservice/generators/microservice
```

File: cli/generator-op-microservice/generators/microservice/templates/src/typedef.graphql [hunk 3]
```diff
@@ -1,7 +1,7 @@
 type Query {
   # Query Definitions
   list: [<%= serviceClassName %>Type]
-  get(ID: ID!): <%= serviceClassName %>Type
+  get(_id: String!): <%= serviceClassName %>Type
```
Comment (deshmukhmayur) [inline cli/generator-op-microservice/generators/microservice/templates/src/typedef.graphql:4]: Avoid using `_` for arg names in graphql.
```suggestion
  get(id: String!): <%= serviceClassName %>Type
```

Comment (deshmukhmayur) [reply cli/generator-op-microservice/generators/microservice/templates/src/typedef.graphql:4]: Same for other queries and. mutations.

**Comment by deshmukhmayur** [comment]
@riginoommen You need to remove the `.test.env` from the webpack.common.js, and add the config to the jest.config.js.

File: cli-init.sh (hunk 1 above)
Comment (riginoommen) [reply cli-init.sh:2]: thats already taken care of

File: cli/generator-op-microservice/generators/microservice/templates/src/typedef.graphql (hunk 3 above)
Comment (riginoommen) [reply cli/generator-op-microservice/generators/microservice/templates/src/typedef.graphql]: Updated

**Comment by riginoommen** [comment]
> @riginoommen You need to remove the `.test.env` from the webpack.common.js, and add the config to the jest.config.js.

In webpack.common.js it defines the env configuration to be selected based on the testing. jest.config.js is nothing to do with it. 

- What happens in webpack.common.ts is whenever the test environment is selected the environment variables are overriden by the ```.test.env``` file. and when the environment swiches back to other it will use actual ```.env``` file.

**Comment by deshmukhmayur** [comment]
@riginoommen But when testing, jest does not use the webpack build, it is running the tests directly on the `.spec.ts` file. Try debugging it by adding console log next to the `envPath = 'e2e/.test.env'`, you'll see that that line is never executed, even when running `npm test`.

**Pull request merged**